### PR TITLE
FEC-12124 Add YouTube referenceId support.

### PR DESCRIPTION
**Pull request by chausov**
Solves FEC-12124
Added YouTube referenceId support.

```Swift
mediaProvider.loadMedia { (mediaEntry, error) in
            
            if let externalSourceType = mediaEntry?.metadata?["externalSourceType"],
               externalSourceType == "YouTube" {
                
                print("YouTube is Playing: \(mediaEntry?.metadata?["referenceId"])")
                
            } else {
                if let me = mediaEntry, error == nil {
                    // create media config
                    let mediaConfig = MediaConfig(mediaEntry: me, startTime: 0.0)
                    
                    // prepare the player
                    self.player!.prepare(mediaConfig)
                }
            }
        }
```

### CODE REVIEW

File: Sources/OVP/Provider/OVPMediaProvider.swift [hunk 1]
```diff
@@ -275,7 +275,6 @@ public enum OVPMediaProviderError: PKError {
                     }
                     
                     guard let baseEntry = ovpBaseEntryList?.objects?.last,
-                        let sources = ovpPlaybackContext?.sources,
```
Comment (x-NR-x) [inline Sources/OVP/Provider/OVPMediaProvider.swift:278]: Why was this removed?
Can we continue the code if we have no source?
The MediaEntry is built on the source.

Comment (chausov) [reply Sources/OVP/Provider/OVPMediaProvider.swift:278]: The reason is that we may have reference ID now (to YouTube video) and no sources in this case will be provided.

File: Sources/OVP/Provider/OVPMediaProvider.swift [hunk 2]
```diff
@@ -329,7 +330,20 @@ public enum OVPMediaProviderError: PKError {
...
+                        if let sources = sources,
+                           baseEntry.externalSourceType != "YouTube" && sources.isEmpty {
+                            PKLog.debug("Response is not containing entry info or playback data.")
+                            callback(nil, OVPMediaProviderError.invalidResponse)
+                            return
+                        }
+                        
+                        metaDataItems["externalSourceType"] = baseEntry.externalSourceType
+                        metaDataItems["referenceId"] = baseEntry.referenceId
+                    }
```
Comment (x-NR-x) [inline Sources/OVP/Provider/OVPMediaProvider.swift]: We need to take care of the case that the baseEntry is **not** an OVPExternalMediaEntry and the sources came back empty.

**Review by x-NR-x — CHANGES_REQUESTED** [review]

File: Sources/OVP/Provider/OVPMediaProvider.swift (hunk 2 above)
Comment (chausov) [reply Sources/OVP/Provider/OVPMediaProvider.swift]: done

**Review by x-NR-x — APPROVED** [review]
